[gladwintan] iP

gladwintan opened the pull request:
# Luna
> "Your mind is for having ideas, not holding them." - David Allen [source](https://dansilvestre.com/productivity-quotes)

A bot to help you organise your tasks in one place. It's,
- Text-based 
- Easy to use 
- ~~FAST~~ SUPER FAST

All you need to do is: 
1. Download it from [here](https://github.com/gladwintan/ip/releases/tag/v0.1)
2. Run the application.
3. Add your tasks. (`todo return book`)
4. And the bot will organise your tasks for you. 

It's that SIMPLE and it is **FREE**. 👍 

Features:
- [x] Organise tasks
- [x] Reminders (coming soon) 

## 
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gladwintan committed:
updated Duke.java to greet user and exit

gladwintan committed:
Rename Duke.java to Luna.java

File is named Duke.java

Changed chatbot name from Duke to Luna and changed class name

gladwintan committed:
Luna.java: add echo function to output user command

Let's
1. add function to output user command and exit when the user types the
	command "bye"

gladwintan committed:
Luna.java: add function to store and display user text

Let's
* update main to store user text
* return user text when requested

gladwintan committed:
Add Task.java

Use of string to represent task input from user.

The task status needs to be modified from undone to done.

Let's add a Task class to represent the tasks from the user.

Having a new class allows us to change the state of the different tasks
easily

gladwintan committed:
Task.java: Add functions to mark and unmark task

gladwintan committed:
Luna.java: Add ability to mark and unmark task

Let's update the main function to enable users to mark and unmark tasks

gladwintan committed:
Add Deadline.java, Event.java, ToDo.java

Require subclass to represent the different tasks

Let's create 3 subclass of Task

Creating subclass allows for use of polymorphism

gladwintan committed:
Luna.java: update main to handle different tasks

gladwintan committed:
Create test for text UI testing

Let's add input for test and expected output to automate the testing of
Luna.

gladwintan committed:
Luna.java: Add error handling

Lack of error handling for incorrect inputs from user

Let's create new exception class to deal with erroneous inputs from user
and update function to catch potential errors

Using new exception class allows us to isolate the error to bad inputs
from user

gladwintan committed:
Luna.java: Add "delete" command

No functionality to delete added tasks

Let's
1. modify the main function to allow users to delete tasks from their
   list of tasks
2. Change how tasks is stored from array to ArrayList

Using ArrayList allows easy deletion and addition of Tasks over array.

gladwintan committed:
Luna.java: Add enum class for command

Let's add enum class with the necessary commands

Enables easy use of switch statement which improves readability and
maintainability of code

gladwintan committed:
Rename ToDo class

gladwintan committed:
Add Storage class to store tasks

gladwintan committed:
Add method to format task for storing

gladwintan committed:
Update chatbot to load and save tasks

gladwintan committed:
Merge branch 'branch-Level-7'

gladwintan committed:
Update deadline task to use LocalDateTime as date

gladwintan committed:
Update event tasks to use LocalDateTime as date

gladwintan committed:
Merge branch 'branch-Level-8'

gladwintan committed:
Add TaskList class

gladwintan committed:
Add Command class and subclasses for different commands

gladwintan committed:
Add Parser class to parse user input

gladwintan committed:
Luna.java: update to use TaskList, Storage, Parser

gladwintan committed:
Add Ui.java for user interactions

gladwintan committed:
Merge branch 'branch-A-MoreOOP'

gladwintan committed:
Create package luna.task

gladwintan committed:
Create luna.command package

gladwintan committed:
Merge branch 'branch-A-Packages'

gladwintan committed:
Merge branch 'add-gradle-support'

gladwintan committed:
Configure gradle

gladwintan committed:
Merge branch 'branch-A-Gradle'

gladwintan committed:
Add JUnit tests for Parser and TaskList

gladwintan committed:
Merge branch 'branch-A-JUnit'

gladwintan committed:
Ui.java: update line shown after each command

gladwintan committed:
Add JavaDoc comments

gladwintan committed:
Update code to comply with coding standard

gladwintan committed:
Add Find command for searching tasks

gladwintan committed:
Merge branch 'branch-A-JavaDoc'

gladwintan committed:
Merge branch 'branch-A-CodingStandard'

gladwintan committed:
Add JavaDoc comments

gladwintan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/luna/Parser.java

gladwintan committed:
Configure Checkstyle

gladwintan committed:
Update code to comply with styling rules

gladwintan committed:
Merge branch 'branch-A-CheckStyle'

gladwintan committed:
Add JavaFX GUI

gladwintan committed:
Command class: update execute to return response

gladwintan committed:
Remove text-based UI

gladwintan committed:
Storage.java: Update loadTask

starrdye commented on src/main/java/luna/Storage.java:
```diff
@@ -0,0 +1,106 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
```
I like how you sorted the import statements.

starrdye commented on src/main/java/luna/Parser.java:
```diff
@@ -0,0 +1,223 @@
+public class Parser {
+    enum CommandType {
+        LIST,
+        TODO,
```
Cool that you created a enum CommandType for the commands.

starrdye commented on src/main/java/luna/Storage.java:
```diff
@@ -0,0 +1,106 @@
+
+            while (sc.hasNextLine()) {
+                String input = sc.nextLine();
+                String[] taskInfo = input.split("\\|");
```
Would it be easier to understand if we use the plural form?

gladwintan committed:
Change user and background image

gladwintan committed:
Update invalid input GUI

gladwintan committed:
Change chat background

gladwintan committed:
Merge branch 'branch-Level-10'

The0nlyJuan commented on src/main/java/luna/task/Todo.java:
```diff
@@ -0,0 +1,27 @@
+/**
+ * Represents a task to be completed.
+ */
+public class Todo extends Task {
```
Todo can be ToDo instead

The0nlyJuan commented on src/main/java/luna/Luna.java:
```diff
@@ -0,0 +1,47 @@
+        try {
+            Command command = Parser.parse(input);
+            return command.execute(tasks, storage);
+
```
do not need a linespace here

The0nlyJuan reviewed (commented):
Overall very clean code LGTM

gladwintan committed:
TaskList.java: Add message for empty list

Message for empty list when list command is entered returns a message
with a list of task.

Let's return a new message if list is empty.

This improves the UI and informs the user of an empty list.

gladwintan committed:
Add assert feature

Assertions are not used to catch potential bugs.

Let's add assertions to Storage and TaskList class.

Both classes handle task manipulation where potential bugs could be
introduced.

gladwintan committed:
Refactor code to improve code quality

Unused imports and unnecessary line breaks found in multiple files.
Name of some variables are not as intuitive.

Let's
* remove unused imports and line breaks in all files
* rename variables

Changing the above would improve readability of code.

gladwintan committed:
Remove text UI

gladwintan committed:
Refactor code

gladwintan committed:
Task class: change class to abstract

gladwintan committed:
Merge pull request #1 from gladwintan/branch-A-Assertions

Branch-A-Assertions

gladwintan committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/luna/Storage.java

gladwintan committed:
Merge pull request #2 from gladwintan/branch-A-CodeQuality

Branch-A-CodeQuality

gladwintan committed:
Add CI

gladwintan committed:
Add function to undo last command

gladwintan committed:
Update undo function

gladwintan committed:
Change Command class to interface

gladwintan committed:
Add Javadoc comments

gladwintan committed:
Merge branch 'branch-BCD-Extension'

gladwintan committed:
Change GUI for invalid commands

gladwintan committed:
Show user's invalid command

gladwintan committed:
Update GUI

gladwintan committed:
Merge branch 'branch-A-BetterGui'

gladwintan committed:
Change file path for storage

gladwintan committed:
Add test for deleting task

gladwintan committed:
Add test for adding tasks

gladwintan committed:
Refactor code

gladwintan committed:
Merge branch 'branch-A-MoreTesting'

gladwintan committed:
Add product screenshot

gladwintan committed:
Add user guide

gladwintan committed:
Refactor luna.txt

gladwintan committed:
Update user guide

gladwintan committed:
Refactor parse method

gladwintan committed:
Update user guide

gladwintan committed:
Remove whitespace from user input
